### PR TITLE
Pattern Grid: Update design of pattern cards

### LINKED ISSUE #103: Pattern Grid: Update patterns with new design
Items should have a fixed height (195px in the design). Title and some meta information (see #100) appear under each pattern. On hover, the favorite & copy actions appear.

### PR DESCRIPTION
**Pull request by ryelle**
Fixes #103 - This updates the UI of the pattern cards to match the new design:
- Title & meta are always visible, only adding favorite count for now since we don't have views
- Show small author avatar
- Show a status label on draft/pending/declined patterns (seen in My Patterns)
- Fix an issue with encoded characters in pattern titles (ex, `&amp;`)
- Fix extra space in the preview — should fix "Extra space on the left/right of full-width items" in #156 

❗  This uses a fake favorite value for now, to get the UI working. I think we'll need a solution with caching to get this value for real, so that discussion can happen in #100 

### Screenshots

Browsing patterns:
all-patterns

My patterns:
my-patterns

Responsive styles - drops to 2 column around 770px, and to 1 col around 525px.
all-patterns-small

all-patterns-mobile

### How to test the changes in this Pull Request:

- Build the branch with `yarn workspaces run build`
- Make sure you have a page called My Patterns (see #152), and a handful of patterns by your user
- View the pattern archives (all and categories)
- They should match the figma design
- View My Patterns
- The patterns should match the design, draft & pending patterns should be labelled
- Resizing the screen shouldn't break anything

### CODE REVIEW

File: public_html/wp-content/themes/pattern-directory/src/components/iframe/index.js [hunk 1]
```diff
@@ -93,14 +93,27 @@ function setHead( doc, head ) {
...
+        width: 100%
+    }
+    </style>`;
 
 	if ( themeSlug ) {
 		headHTML += `<link rel="stylesheet" href="https://wp-themes.com/wp-content/themes/${ themeSlug }/style.css" media="all" />`;
 	} else {
 		headHTML +=
-			'<link rel="stylesheet" href="https://wp-themes.com/wp-content/themes/twentytwentyone/style.css" media="all" />';
+			'<link rel="stylesheet" href="https://wp-themes.com/wp-content/themes/twentynineteen/style.css" media="all" />';
```
Comment (StevenDufresne) [inline public_html/wp-content/themes/pattern-directory/src/components/iframe/index.js]: This isn't mentioned in the description. Is this on purpose?

File: public_html/wp-content/themes/pattern-directory/src/components/pattern-thumbnail/index.js [hunk 2]
```diff
@@ -1,29 +1,65 @@
...
+				) : null }
+				<div className="pattern-grid__actions">
+					<FavoriteButton showLabel={ false } patternId={ pattern.id } />
+					<CopyPatternButton isSmall={ true } content={ pattern.pattern_content } />
+				</div>
 			</div>
+
+			<h2 className="pattern-grid__title">{ decodeEntities( pattern.title.rendered ) }</h2>
+			<p className="pattern-grid__meta">
+				{ showAvatar && pattern.author_avatar ? (
```
Comment (StevenDufresne) [inline public_html/wp-content/themes/pattern-directory/src/components/pattern-thumbnail/index.js]: Should there be a default case for missing authors... if that case is possible??

File: public_html/wp-content/themes/pattern-directory/src/components/pattern-thumbnail/index.js [hunk 3]
```diff
@@ -1,29 +1,65 @@
...
-					<Canvas className="pattern-grid__preview" html={ pattern.content.rendered } />
-				</Disabled>
-			</a>
-			<div className="pattern-grid__actions">
-				<h2 className="pattern-grid__title">{ pattern.title.rendered }</h2>
-				<FavoriteButton showLabel={ false } patternId={ pattern.id } />
-				<CopyPatternButton isSmall={ true } content={ pattern.pattern_content } />
+			<div className="pattern-grid__pattern-frame">
+				<a href={ pattern.link } rel="bookmark">
+					<span className="screen-reader-text">{ decodeEntities( pattern.title.rendered ) }</span>
```
Comment (StevenDufresne) [inline public_html/wp-content/themes/pattern-directory/src/components/pattern-thumbnail/index.js:34]: I can't remember, but are we relying on `.org` to have this style globally?

**Review by StevenDufresne — APPROVED** [review]
A few minor comments. Looks good.

File: public_html/wp-content/themes/pattern-directory/src/components/iframe/index.js (hunk 1 above)
Comment (StevenDufresne) [reply public_html/wp-content/themes/pattern-directory/src/components/iframe/index.js]: I now see this mentioned in #177.

File: public_html/wp-content/themes/pattern-directory/src/components/pattern-thumbnail/index.js (hunk 3 above)
Comment (ryelle) [reply public_html/wp-content/themes/pattern-directory/src/components/pattern-thumbnail/index.js:34]: Do you mean the `screen-reader-text`? it's included in [the parent theme's CSS](https://meta.trac.wordpress.org/browser/sites/trunk/wordpress.org/public_html/wp-content/themes/pub/wporg/css/objects/_accessibility.scss), so it's bundled into the built styles.

**Comment by shaunandrews** [comment]
I think we should link the pattern title to view the pattern. Similarly, the author's avatar should be linked to their -archive page.

--

It'd be nice if you could click the heart icon to favorite the pattern. I know its a duplicate action, but it feels strange to have the icon _not_ be interactive.

--

> only adding favorite count for now since we don't have views

Will we have views? If not, perhaps we can display the authors name?

--

image

I'm seeing teeny-tiny scrollbars, and many patterns are rendered outside their container — even when viewing on the detail page.

--

image

The Favorite button's `:focus` state is a single pixel line without any border-radius. Instead, it should match the copy button's focus state, using a thicker border with the border-radius.

--

On My Patterns I'm able to see a "Draft" indicator:

image

In the designs, the text appears bolder and there's a border-radius applied to the shape:

image

Also, maybe the badges should have `pointer-events: none;` to that it doesn't block the ability to click the pattern or allow people to select it's text.

File: public_html/wp-content/themes/pattern-directory/src/components/pattern-thumbnail/index.js (hunk 3 above)
Comment (StevenDufresne) [reply public_html/wp-content/themes/pattern-directory/src/components/pattern-thumbnail/index.js:34]: 👍

**Comment by shaunandrews** [comment]
I've still got this branch running and noticed that the copy button has a nice, subtle hover state — but the favorite button doesn't. Can we add a hover state for the favorite button:

image

**Comment by ryelle** [comment]
Okay, latest changes include:
- Author display name is shown, and links to the author page (author filter doesn't work yet though)
- Title links to pattern
- Favorite count is now also a favorite button
- Removed the scrollbars from the small previews
- Fixed the style on the status labels, no longer block clicks

A favorited pattern with favorite focused:
Screen Shot 2021-06-16 at 5 20 32 PM

A pending pattern with favorite hovered:
Screen Shot 2021-06-16 at 5 20 04 PM

^ The above don't have authors because it's the My Patterns page. From the main archive:
Screen Shot 2021-06-16 at 5 19 00 PM

**Comment by ryelle** [comment]
Merging this, any followup feedback can be addressed in issues.

**Pull request merged**